cloudblue_connect.custom_adapter.rb

```diff
@@ -0,0 +1,694 @@
+#
+# This file is a part of the Ingram Micro’s CloudBlue Connect Workato Extension.
+
+#
+
+# @copyright (c) 2021. Ingram Micro. All Rights Reserved.
+
+#
+
+
+{
+  title: "CloudBlue Connect",
+
+  # API key authentication example. See more examples at https://docs.workato.com/developing-connectors/sdk/guides/authentication.html
+  connection: {
+    fields: [
+      {
+        name: "api_token",
+        label: "API Token",
+        hint: "You can find your API Token in <a href='https://portal.connect.cloudblue.com/integrations/tokens/' target='_blank'>Connect</a>",
+        optional: false
+      },
+    ],
+
+    authorization: {
+      type: "custom_auth",
+
+      # Calendly uses header auth (Authorization: <api key>)
+      apply: lambda do |connection|
+        headers("Authorization": connection["api_token"])
+      end
+    },
+
+    base_uri: lambda do
+      "https://api.connect.cloudblue.com"
+    end
+  },
+
+  test: lambda do |_connection|
+    get("/public/v1")
+  end,
+
+  object_definitions: {
+  
+    cancel_request: {
+      fields: lambda do
+        [
+          { name: "id"},
+          { name: "type"},
+          { name: "error_code"},
+          { name: "status" },
+          { name: "params_form_url"},
+          { name: "asset", type: "object", properties: [
+            { name: "id" },   
+            { name: "external_id" }
+          ]},
+          { name: "errors", type: "array"},
+        ]
+      end
+  
+    },    
+    purchase_request: {
+      fields: lambda do
+        [
+          { name: "id"},
+          { name: "type"},
+          { name: "status" },
+          { name: "params_form_url"},
+          { name: "asset", type: "object", properties: [
+            { name: "id" },   
+            { name: "external_id" }
+          ]}
+        ]
+      end
+  
+    },
+    change_request: {
+      fields: lambda do
+        [
+          { name: "id"},
+          { name: "type"},
+          { name: "error_code"},
+          { name: "status"},
+          { name: "asset", type: "object", properties: [
+            { name: "id"},
+            { name: "external_id"},
+            { name: "external_uid"}
+            ]
+          },
+          { name: "errors", type: "array"},
+          
+          
+         ]
+      end
+    },
```

Comment:
these structures look very similar. suggestion is to replace them with a single structure 'created_request'